src/asm.rs

```diff
@@ -4,4 +4,8 @@
 
 use core::arch::global_asm;
 
+#[cfg(not(test))]
 global_asm!(include_str!("start.S"));
+
+#[cfg(test)]
+global_asm!(include_str!("test_start.S"));
```

Comment:
do we really need a separate `start.S`?

Reply:
I want to make sure that there is no way for testing code to end up in the shipping binary, and there is no way to use cfg in inline assembly.